service/matching/tasklist/task_list_manager.go

```diff
@@ -240,7 +240,7 @@ func NewManager(
 			partitionConfig := tlMgr.TaskListPartitionConfig()
 			r := 1
 			if partitionConfig != nil {
-				r = int(partitionConfig.NumReadPartitions)
+				r = len(partitionConfig.ReadPartitions)
```

Comment:
This `len()` logic is reused at multiple places, let's consider creating a helper method.